### ISSUE TITLE
Frequent error in build-datasets script

### ISSUE THREAD

**Issue by mmarchegiani** [PocketCoffea/PocketCoffea#259]
When running the `pocket-coffea build_datasets` script with parallel processes, a frequent error related to the JSON decoder is raised:
```
"""
Traceback (most recent call last):
  File "/work/mmarcheg/PocketCoffea/pocket_coffea/utils/dataset.py", line 32, in do_dataset
    dataset = Dataset(
  File "/work/mmarcheg/PocketCoffea/pocket_coffea/utils/dataset.py", line 239, in __init__
    self.get_samples(self.cfg["files"])
  File "/work/mmarcheg/PocketCoffea/pocket_coffea/utils/dataset.py", line 254, in get_samples
    sample = Sample(
  File "/work/mmarcheg/PocketCoffea/pocket_coffea/utils/dataset.py", line 117, in __init__
    self.get_filelist()
  File "/work/mmarcheg/PocketCoffea/pocket_coffea/utils/dataset.py", line 159, in get_filelist
    files_replicas, sites, sites_counts = rucio.get_dataset_files_replicas(
  File "/work/mmarcheg/PocketCoffea/pocket_coffea/utils/rucio.py", line 169, in get_dataset_files_replicas
    sites_xrootd_prefix = get_xrootd_sites_map()
  File "/work/mmarcheg/PocketCoffea/pocket_coffea/utils/rucio.py", line 94, in get_xrootd_sites_map
    return json.load(open(".sites_map.json"))
  File "/work/mmarcheg/micromamba/envs/pocket-coffea/lib/python3.9/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/work/mmarcheg/micromamba/envs/pocket-coffea/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/work/mmarcheg/micromamba/envs/pocket-coffea/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/work/mmarcheg/micromamba/envs/pocket-coffea/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

This is possibly due to multiple processes trying to write to the same file at the same time. We need to fix this and avoid processes from conflicting with each other.

**Comment by andreypz** [comment]
Agreed, this one is annoying.
I think the solution should be simply avoiding write/read of the `sites_map.json` file altogether. Instead, this dictionary could be created on the fly by each process every time, and just kept in memory. I think it does not take much time to re-create.

**Comment by mmarchegiani** [comment]
Solved in #261
